homeassistant/components/waze_travel_time/sensor.py

```diff
@@ -67,6 +87,39 @@ async def async_setup_entry(
     sensor = WazeTravelTime(config_entry.entry_id, name, origin, destination, data)
 
     async_add_entities([sensor], False)
+    if not hass.services.has_service(DOMAIN, SERVICE_UPDATE_TRAVEL_TIME):
+        async_setup_services(hass)
+
+
+@callback
+def async_setup_services(hass: HomeAssistant) -> None:
+    """Set up services for waze_travel_time integration."""
+
+    platform = entity_platform.async_get_current_platform()
+
+    async def async_update_travel_time(service_call: ServiceCall) -> ServiceResponse:
+        assert platform is not None
```

Comment:
As implemented, this looks like it should be an entity service https://developers.home-assistant.io/docs/dev_101_services/#entity-services -- however I think to make that work, we may need to update it's registration method to support this if that is the right direction to go, like I had to do for calendar list events for the entity component.

Reply:
I tried to implement it with an entity_service until I realized it does not support response values. In the long run I want this service to live in a "base" component just like you mentioned for calendar list events. The new base component could be `travel_time`.

I saw that `calendar` uses `component.async_register_entity_service` like other components. Is this a way which is also allowed for "downstream" components like `waze`,`google`,`here` or restricted to base components like `light`,`calendar`,`button`,etc. ?

Reply:
https://github.com/home-assistant/core/pull/96370